packages/url/src/index.js

```diff
@@ -159,19 +159,33 @@ export function isValidFragment( fragment ) {
 }
 
 /**
- * Appends arguments to the query string of the url
+ * Appends arguments as querystring to the provided URL. If the URL already
+ * includes query arguments, the arguments are merged with (and take precedent
+ * over) the existing set.
  *
- * @param {string} url  URL
- * @param {Object} args Query Args
+ * @param {?string} url  URL to which arguments should be appended. If omitted,
+ *                       only the resulting querystring is returned.
+ * @param {Object}  args Query arguments to apply to URL.
  *
- * @return {string} Updated URL
+ * @return {string} URL with arguments applied.
  */
-export function addQueryArgs( url, args ) {
+export function addQueryArgs( url = '', args ) {
+	let baseUrl = url;
+
+	// Determine whether URL already had query arguments.
 	const queryStringIndex = url.indexOf( '?' );
-	const query = queryStringIndex !== -1 ? parse( url.substr( queryStringIndex + 1 ) ) : {};
-	const baseUrl = queryStringIndex !== -1 ? url.substr( 0, queryStringIndex ) : url;
+	if ( queryStringIndex !== -1 ) {
+		// Merge into existing query arguments.
+		args = Object.assign(
+			parse( url.substr( queryStringIndex + 1 ) ),
```

Comment:
[MDN recommends using `substring` over `substr`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr), as `substr` isn't actually part of the JS language spec.

I'm sure it's not the only place we use it, but just saying' 🤷‍♂️

Reply:
> [MDN recommends using `substring` over `substr`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr), as `substr` isn't actually part of the JS language spec.

TIL! In this instance it would have been perfectly fine to replace the usage. I have found in some cases the behavior difference of providing `length` as an argument to be of some convenience, though the fact that it's not standardized is reason enough to want to avoid it.